Add io.github.DenysMb.Kontainer

DenysMb opened the pull request:
<!-- ⚠️⚠️ Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Go to the preview tab to click the links below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

- [X] Please describe the application briefly. <!-- insert the description here --> **A simple Kirigami GUI for Distrobox**
- [X] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor to the project. [Repo](https://github.com/DenysMb/Kontainer)

<!-- 💡 Mention any additional maintainers needed below 💡 -->

<!-- ⚠️⚠️ DO NOT modify anything below this line ⚠️⚠️  -->

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

hfiguiere commented on io.github.DenysMb.Kontainer.json:
```diff
@@ -0,0 +1,32 @@
+    "finish-args": [
+        "--share=ipc",
+        "--socket=wayland",
+        "--socket=session-bus",
```
this is not allowed. you need to determine which bus name you need and list them explicitly.

DenysMb replied:
I realize it was unnecessary. Removed.

hfiguiere commented on io.github.DenysMb.Kontainer.json:
```diff
@@ -0,0 +1,32 @@
+    "command": "kontainer",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=wayland",
```
so what about non wayland session?

DenysMb replied:
Just added "fallback-x11" too.

hfiguiere commented on io.github.DenysMb.Kontainer.json:
```diff
@@ -0,0 +1,32 @@
+        "--socket=wayland",
+        "--socket=session-bus",
+        "--device=dri",
+        "--env=DISPLAY=:0",
```
not sure why you want to set DISPALY while not allowing X11, but both are incorrect.

DenysMb replied:
No need for this. Removed.

hfiguiere commented on io.github.DenysMb.Kontainer.json:
```diff
@@ -0,0 +1,32 @@
+        "--device=dri",
+        "--env=DISPLAY=:0",
+        "--filesystem=home",
+        "--talk-name=org.freedesktop.Flatpak"
```
this is not allowed by default. You need to justify it and then apply for an exception.

DenysMb replied:
The reason for use this is because I need to run the distrobox from system, so I ran with `flatpak-spawn --host distrobox`. Is there a better way so I don't need to use this `--talk-name=org.freedesktop.Flatpak` here?

bbhtt replied:
submit an exception https://docs.flathub.org/docs/for-app-authors/linter#exceptions

https://docs.flathub.org/docs/for-app-authors/linter#finish-args-flatpak-spawn-access

DenysMb replied:
PR created: https://github.com/flathub-infra/flatpak-builder-lint/pull/640

hfiguiere commented on io.github.DenysMb.Kontainer.json:
```diff
@@ -0,0 +1,32 @@
+                {
+                    "type": "git",
+                    "url": "https://github.com/DenysMb/Kontainer.git",
+                    "branch": "main"
```
never a git branch

DenysMb replied:
Changed. Using a tag now.

DenysMb commented on io.github.DenysMb.Kontainer.json:
```diff
@@ -0,0 +1,32 @@
+        "--socket=session-bus",
+        "--device=dri",
+        "--env=DISPLAY=:0",
+        "--filesystem=home",
```
Removed. I forgot to remove this before. No need since the distrobox that we access is the system distrobox (that already has access to home).

hfiguiere replied:
still not removed.

DenysMb replied:
Removed now, sorry.

hfiguiere reviewed (changes requested):
I don't see any appstream file being installed.

Nor properly named icons.

(this is part of the requirements)

DenysMb commented:
> I don't see any appstream file being installed.
> 
> Nor properly named icons.
> 
> (this is part of the requirements)

Renamed the icon to **io.github.DenysMb.Kontainer.svg** and added **io.github.DenysMb.Kontainer.metainfo.xml**.

I can find both in `.../export/share/icons/hicolor/scalable/apps/io.github.DenysMb.Kontainer.svg` and `.../export/share/metainfo/io.github.DenysMb.Kontainer.metainfo.xml`

bbhtt commented:
bot, build

DenysMb commented:
Strange. I was able to build for x86, but pipeline don't even ran for x86 because it fails for aarch64.
I copied the `exceptions.json` from [flatpak-builder-lint](https://github.com/flathub-infra/flatpak-builder-lint/blob/master/flatpak_builder_lint/staticfiles/exceptions.json) repository and ran `flatpak run --command=flatpak-builder-lint org.flatpak.Builder --exceptions --user-exceptions exceptions.json manifest io.github.DenysMb.Kontainer.json`. No error showed.

hfiguiere commented:
if one pipeline fails the other is cancelled. Check the errors in the build log before calling foul.

DenysMb commented:
Changed screenshot URLs on metadata file and added homepage URL and content_rating.
Lint passed fine for manifest and appstream. Everything should be fine now.

hfiguiere commented on io.github.DenysMb.Kontainer.json:
```diff
@@ -23,7 +23,7 @@
                     "url": "https://github.com/DenysMb/Kontainer.git",
                     "tag": "1.0.0",
-                    "commit": "238c0abcd969ce32075aeca0abff032723853165"
+                    "commit": "fd7de4b15e846b4ee5c46ff6fa935b9a28771cc6"
```
retagging is a bad practice. And that's not flatpak specific.

DenysMb replied:
I know, I usually don't do this, and I could avoid if I didn't choose to make a tag only when making a release.
Since the changes were only on the manifest and metadata, not being reflected on the application, my preference was to re-tag instead of doing a 1.0.X release.
With everything running fine now the tags and release will follow the semantic versioning.

hfiguiere commented:
bot, build

bbhtt commented:
/merge
